client/src/main/java/io/split/inputValidation/ImpressionPropertiesValidator.java

```diff
@@ -0,0 +1,83 @@
+package io.split.inputValidation;
+
+import io.split.client.dtos.KeyImpression;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class ImpressionPropertiesValidator {
+    private static final Logger _log = LoggerFactory.getLogger(ImpressionPropertiesValidator.class);
+
+    public static ImpressionPropertiesValidatorResult propertiesAreValid(Map<String, Object> properties) {
+        int size = 1024; // We assume 1kb events without properties (750 bytes avg measured)
+
+        if (properties == null) {
+            return new ImpressionPropertiesValidatorResult(true);
+        }
+        if (properties.size() > 300) {
+            _log.warn("Impression properties has more than 300 properties. Some of them will be trimmed when processed");
+        }
+
+        Map<String, Object> result = new HashMap<>();
+        for (Map.Entry<String, Object> entry : properties.entrySet()) {
+            if (entry.getKey() == null || entry.getKey().isEmpty()) {
+              continue;
+            }
+
+            size += entry.getKey().length();
+            Object value = entry.getValue();
+
+            if (!(value instanceof Number) && !(value instanceof Boolean) && !(value instanceof String)) {
+                _log.warn(String.format("Property %s is of invalid type. Setting value to null", entry.getKey()));
+                value = null;
+            }
+
+            if (value instanceof String) {
+                size += ((String) value).length();
+            }
+
+            if (size > KeyImpression.MAX_PROPERTIES_LENGTH_BYTES) {
+                _log.error(String.format("The maximum size allowed for the properties is 32768 bytes. "
+                        + "Current one is %s bytes. Properties field is ignored", size));
+
+                return new ImpressionPropertiesValidatorResult(false);
+            }
+
+            result.put(entry.getKey(), value);
+        }
+
+        return new ImpressionPropertiesValidatorResult(true, size, result);
+    }
```

Comment:
These validations are exactly the same as the ones [here](https://github.com/splitio/java-client/blob/master/client/src/main/java/io/split/inputValidation/EventsValidator.java), if I'm correct. Can we improve this implementation and avoid duplicating code?

Reply:
will do